lib/ld-key-pair.js

```diff
@@ -207,6 +221,43 @@ class Ed25519KeyPair extends LDKeyPair {
     const privateKey = JSON.parse(decipher.output.getBytes());
     return privateKey;
   }
+
+  /**
+   * Generates and returns a Multiformat encoded ed25519 public key fingerprint
+   * (for use with cryptonyms, for example).
+   *
+   * @see https://github.com/multiformats/multicodec
+   *
+   * @returns {string}
+   */
+  fingerprint() {
+    const forge = this.injector.use('node-forge');
+    const buffer = new forge.util.createBuffer();
+
+    // ed25519 cryptonyms are multiformat encoded values, specifically:
+    // (multicodec 0x30 + ed25519-pub 0xed + key bytes)
+    const pubkeyBytes = forge.util.binary.base58.decode(this.publicKey);
+    buffer.putBytes(forge.util.hexToBytes('30ed'));
+    buffer.putBytes(pubkeyBytes.toString('binary'));
+
+    return forge.util.binary.base58.encode(buffer);
+  }
+
+  /**
+   * Tests whether the fingerprint was generated from a given key pair.
+   *
+   * @param fingerprint {string}
+   *
+   * @returns {boolean}
+   */
+  verifyFingerprint(fingerprint) {
+    const forge = this.injector.use('node-forge');
+    const fingerprintBuffer = forge.util.binary.base58.decode(fingerprint);
+    const publicKeyBuffer = forge.util.binary.base58.decode(this.publicKey);
+
+    // slice off the two first multiformat bytes, 0x30 and 0xed
+    return publicKeyBuffer.equals(fingerprintBuffer.slice(2));
```

Comment:
has the information being discarded here already been validated?  Does it need to be?